airflow/providers/fab/auth_manager/fab_auth_manager.py

```diff
@@ -159,15 +159,15 @@ def get_api_endpoints(self, connexion_app: connexion.FlaskApp) -> None | Bluepri
             swagger_ui=conf.getboolean("webserver", "enable_swagger_ui", fallback=True),
         )
 
-        api = connexion_app.add_api(
+        connexion_app.add_api(
             specification=specification,
             resolver=_LazyResolver(),
             base_path="/auth/fab/v1",
             swagger_ui_options=swagger_ui_options,
             strict_validation=True,
             validate_responses=True,
         )
-        return api.blueprint if api else None
+        return None
```

Comment:
```suggestion
        self.appbuilder.app.extensions["csrf"].exempt(api.blueprint)
        return None
```
According to the proposed solution, you it is suggested to move it here I guess.

Reply:
This doesn't give you KeyError "crsf"?

Reply:
yeah, you have to add an `if` statement. But this does not solve the CSRF error

Reply:
```
        if api:
            self.appbuilder.app.extensions["csrf"].exempt(api.blueprint)
        return None```

Reply:
Looks like this connexion_app.add_api() doesn't return an object, so I just removed "if" statement.

I kind of understanding Vlada's struggle now.

Reply:
Yes, I think that is why, CSRF error is still there. In this comment I think Rob suggested some solution
https://github.com/apache/airflow/pull/36052#issuecomment-1958422648

Reply:
I'm not sure what he meant here. I might need some time to understand it.

"Find a way to add csrf extension to a newly created blueprint using connexion: to retrieve blueprint object from connexion_app variable to save the current logic (flask_app.extensions["csrf"].exempt(blueprint)) or find a way to add this extension on connexion level(check the documentation for available options)."

Seems like we cannot get the blueprint, so we better focus on the second option(add this extension on the connexion level). I'm reading the connexion source code, but no luck yet.

Reply:
Probably need to understand [this repo](https://github.com/simonw/asgi-csrf/tree/main). I will let you know if I find something useful.